src/h2olog/h2olog.cc

```diff
@@ -125,6 +126,44 @@ static void show_process(pid_t pid)
     infof("Attaching pid=%d (%s)", pid, cmdline);
 }
 
+static void drop_root_privilege(void)
+{
+    if (getuid() == 0) {
+        if (getgid() == 0) {
```

Comment:
I think it might be possible remove this `if`, is it necessary? I think that the behavior we want is "if running as root, drop root privileges."